platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/admin/ImportMigrationConfigurationAdminEntry.java

```diff
@@ -0,0 +1,183 @@
+
+import javax.annotation.Nullable;
+
+import org.apache.commons.lang.Validate;
+import org.codice.ddf.migration.ImportMigrationEntry;
+import org.codice.ddf.migration.ImportMigrationEntryProxy;
+import org.codice.ddf.migration.MigrationException;
+import org.osgi.framework.Constants;
+import org.osgi.service.cm.Configuration;
+import org.osgi.service.cm.ConfigurationAdmin;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class extends on the {@link ImportMigrationEntry} interface to represent an exported
+ * configuration object.
+ */
+public class ImportMigrationConfigurationAdminEntry extends ImportMigrationEntryProxy {
+    private static final Logger LOGGER = LoggerFactory.getLogger(
+            ImportMigrationConfigurationAdminEntry.class);
+
+    private final ImportMigrationConfigurationAdminContext context;
+
+    private final Dictionary<String, Object> properties;
+
+    @Nullable
+    private final String factoryPid;
+
+    private final String pid;
+
+    @Nullable
+    private final Configuration memoryConfiguration;
+
+    private boolean restored = false;
+
+    public ImportMigrationConfigurationAdminEntry(ImportMigrationConfigurationAdminContext context,
+            ImportMigrationEntry entry, Dictionary<String, Object> properties) {
+        super(entry);
+        Validate.notNull(context, "invalid null context");
+        Validate.notNull(properties, "invalid null properties");
+        this.context = context;
+        this.properties = properties;
+        // note: we also remove factory pid and pid from the dictionary as we do not want to restore those later
+        this.factoryPid = Objects.toString(properties.remove(ConfigurationAdmin.SERVICE_FACTORYPID),
+                null);
+        this.pid = Objects.toString(properties.remove(Constants.SERVICE_PID), null);
+        this.memoryConfiguration = context.getMemoryConfiguration(this);
+        // TODO: InterpolationHelper.performSubstitution(strMap, context); where context is a BundleContext
+    }
+
+    @Override
+    public boolean restore() {
+        if (!restored) {
+            this.restored = false; // until proven otherwise
+            final Configuration cfg;
+
+            if (memoryConfiguration != null) {
+                if (propertiesMatch()) {
+                    LOGGER.debug("Importing configuration for [{}] from [{}]; no update required",
+                            memoryConfiguration.getPid(),
+                            getPath());
+                    this.restored = true;
+                    return true;
+                }
+                cfg = memoryConfiguration;
+                LOGGER.debug(
+                        "Importing configuration for [{}] from [{}]; updating existing configuration...",
+                        memoryConfiguration.getPid(),
+                        getPath());
+            } else {
+                if (LOGGER.isDebugEnabled()) {
+                    if (isManagedServiceFactory()) {
```

Comment:
✏️ Nit: Can we put these logs into a private method? Not a big deal, but it really clogs up this method.